File: crates/wbw_r/examples/osm_download_vector.R
# Download OSM vector features with wbw_r.
# Example: fetch road centerlines in a small bbox and write to GeoJSON.

library(whiteboxworkflows)

session <- wbw_session()

result <- wbw_run_tool(
  "download_osm_vector",
  args = list(
    west = -80.54,
    south = 43.41,
    east = -80.47,
    north = 43.47,
    filter_preset = "roads",
    include_points = FALSE,
    include_lines = TRUE,
    include_polygons = FALSE,
    timeout_seconds = 30,
    max_elements = 50000,
    output = "kitchener_roads.geojson"
  ),
  session = session
)

cat(sprintf("wrote output: %s\n", result$outputs$path))

File: crates/wbw_python/examples/osm_download_vector.py
"""Download OSM vector features with wbw_python.

Example: fetch trail centerlines using a projected Web Mercator bbox,
transforming query extents to EPSG:4326 internally.
"""

from wbw import WbEnvironment


def main() -> None:
    wbe = WbEnvironment()

    result = wbe.run_tool(
        "download_osm_vector",
        {
            "west": -8983000.0,
            "south": 5382000.0,
            "east": -8965000.0,
            "north": 5393000.0,
            "input_extent_epsg": 3857,
            "filter_preset": "trails",
            "overpass_profile": "kumi",
            "include_points": False,
            "include_lines": True,
            "include_polygons": False,
            "timeout_seconds": 30,
            "max_elements": 50000,
            "output": "kitchener_trails.geojson",
        },
    )

    print(f"wrote output: {result.outputs.get('path')}")


if __name__ == "__main__":
    main()
